### DISCUSSION TITLE
How reanalyze works after "Set Register Values"?

### DISCUSSION THREAD

**Discussion by dotCirill** [NationalSecurityAgency/ghidra#6298] [Q&A]
When set register at the beginning of the function it works as I expect:

![image](https://github.com/NationalSecurityAgency/ghidra/assets/33807714/0e658508-b51b-487c-83db-ec258440b187)

When set register inside the function decompiler view doesn't update:

![image](https://github.com/NationalSecurityAgency/ghidra/assets/33807714/4fddc479-cda3-4efd-bc40-b059636788c1)

This looks pretty strange for me

**Comment by KaiEpic** [comment]
definetly annoying, have it on many files i analyse now..
